backends/arm/operators/op_mean_dim.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import List
+
+import serializer.tosa_serializer as ts
+import torch
+from executorch.backends.arm.operators.node_visitor import (
+    NodeVisitor,
+    register_node_visitor,
+)
+from executorch.backends.arm.operators.op_common import build_avg_pool_2d_common
+from executorch.backends.arm.tosa_mapping import TosaArg
+
+
+@register_node_visitor
+class MeanDimVisitor(NodeVisitor):
+    target = "aten.mean.dim"
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def define_node(
+        self,
+        node: torch.fx.Node,
+        tosa_graph: ts.TosaSerializer,
+        inputs: List[TosaArg],
+        output: TosaArg,
+        is_quant_node: bool,
+    ) -> None:
+
+        input_tensor = inputs[0]
+        dim = node.args[1]
+        keep_dim = node.args[2]
+
+        # mean.dim(-1, -2) is the same as avg_pool2d when just computing mean over HW dimensions.
+        # Since tosa doesn't have mean.dim operation, lowers it to average pooling instead.
+        if dim == [-1, -2]:
+            if keep_dim is True:
+                # Given the shape format of input is (N, C, H, W)
+                kernel_size = [input_tensor.shape[2], input_tensor.shape[3]]
+                stride = [1, 1]
+                padding = [0, 0, 0, 0]
+
+                build_avg_pool_2d_common(
+                    node,
+                    tosa_graph,
+                    input_tensor,
+                    kernel_size,
+                    stride,
+                    padding,
+                    is_quant_node,
+                    output,
+                )
+                return
+
+        raise AssertionError("unsupported")
```

Comment:
The problem with this approach is that the partitioner call always fails when it reaches here. And the lowering doesn't get a chance to generate a PTE which can fallback to running such ops on Portable. We should rather not partition such nodes in the first place.

Reply:
Make sense. But for the existing model, such as mobilenetv2, that already have aten.mean.dim node, I don't know how to avoid partition that node in the first place? Do you mean add a logic in `TOSASupportedOperators` to report false whenever unsupported cases occur?

Reply:
Yea, I think Digant means adding some logic here to not partition such nodes: https://github.com/pytorch/executorch/blob/dc5a9af392f0104247efe590c8fc5aef76e55137/backends/arm/arm_partitioner.py#L50

Reply:
Yeah exactly - something like [this](https://github.com/pytorch/executorch/blob/main/backends/xnnpack/partition/xnnpack_partitioner.py#L236-L242) - we do in XNNPACK.